app/src/main/java/nyc/ignitelabs/civicduty/view/MainActivity.kt

```diff
@@ -0,0 +1,41 @@
+package nyc.ignitelabs.civicduty.view
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.*
+import nyc.ignitelabs.civicduty.Constants
+import nyc.ignitelabs.civicduty.Constants.CODE_REQUEST_ADDRESS
+import nyc.ignitelabs.civicduty.R
+import nyc.ignitelabs.civicduty.addressEditIntent
+
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_main)
+
+        button_submit.setOnClickListener {
```

Comment:
This is so much cleaner than the `findViewById` path; is it considered idiomatic?

Reply:
So this is possible only with Kotlin extensions for Android. This is the line that enables it: `import kotlinx.android.synthetic.main.activity_edit.*`

There is a library that's used explicitly for this, which is called `ButterKnife` and which LUNA project uses but it's essentially enabling this very similar functionality for our purposes so I just skipped it in favor of this. It's super neat and ability to do s**t like this is why Android devs get excited about using Kotlin!